scripts/core/exp-multi-device-compile.yml

```diff
@@ -93,7 +93,7 @@ analogue:
     - "**clLinkProgram**"
 details:
     - "The application may call this function from simultaneous threads."
-    - "Following a successful call to this entry point the program returned in `phProgram` will contain a binary of the $X_PROGRAM_BINARY_TYPE_EXECUTABLE type for each device in `phDevices`."
+    - "Following a successful call to this entry point the program returned in `phProgram` will contain a binary of the $X_PROGRAM_BINARY_TYPE_EXECUTABLE type for each device in `phDevices`. If a non-success code is returned and `phProgram` is not `nullptr`, `phProgram` will contain an unspecified binary or `nullptr`."
```

Comment:
I wonder if we should be more explicit about the state of the `phProgram`, i.e. that it would be valid to query its build log?

Reply:
We only have 3 error codes defined for this entry point, would if make sense to limit this behaviour to only when `UR_RESULT_ERROR_PROGRAM_LINK_FAILURE` is returned?

Reply:
I've updated the documentation to specifically call out that implementations may use the log file of the program to indicate linker failures. However, for consistency's sake, I'll just have all errors require it being set to nullptr (the work is already done anyway).